back-end/utils/top-layer/PhanKiet.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * 
+ * This file is here because I don't know where should I put these codes in.
+ * Please tell me its suitable location so I can transfer these lines to an appropriate place.
+ * 
+ */
+
+
+const { isEmpty } = require("lodash");
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+
+const { FINANCIAL_MODELING_PREP_API_KEY } = process.env;
+
+
+  
+const getSingleShareRating = (shareSymbolString) =>
+{
+    return new Promise((resolve, reject) =>
+    {
+        fetch(`https://financialmodelingprep.com/api/v3/rating/${shareSymbolString}?apikey=${FINANCIAL_MODELING_PREP_API_KEY}`)
+        .then((stockRating) =>
+        {
+            return stockRating.json();
+        })
+        .then((stockRatingJSON) =>
+        {
+            if (isEmpty(stockRatingJSON))
+            {
+                reject(new Error(`Share symbols do not exist in FMP.`));
+            }
+            else
+            {
+                resolve(stockRatingJSON[0]);
+            }
+        });
+    });
+}
+
+const getStockScreener = (totalCompanies) =>
+{
+    return new Promise((resolve, reject) =>
+    {
+        fetch(
+            `https://financialmodelingprep.com/api/v3/stock-screener?limit=${totalCompanies}&apikey=${FINANCIAL_MODELING_PREP_API_KEY}`
+        )
+        .then((stockRatingsArray) =>
+        {
+            return stockRatingsArray.json();
+        })
+        .then((stockRatingsArrayJSON) =>
+        {
+            resolve(stockRatingsArrayJSON);
+        })
+    });
+}
+
+const getAllSharesRatings = () =>
+{
+    return new Promise((resolve, reject) =>
+    {
+        try
+        {
+            const StockScreener = getStockScreener(600);
+            // eslint-disable-next-line prefer-const
+            let ratings = [];
+            
+            for (let i = 0; i < Object.values(StockScreener).length; ++ i)
+            {
+                ratings.push(getSingleShareRating(StockScreener[i].symbol));
+            }
+
+            resolve(ratings);
+        }
+        catch (err)
+        {
+            reject(err);
+        }
+    });
+}
+
+const updateCompaniesRatingsList = (user) => {
```

Comment:
- Cái này sửa lại hết logic. Mình sẽ không lưu riêng data trong từng user. Cái companies ratings là dùng chung cho toàn bộ server. Nên hồi nãy em có nói là không cần relation user trong model CompanyRating. 
- Đối với function updateCompaniesRatingsList này, anh update thẳng từng item companyRating trong database không cần thông qua user hay gì hết.